drivers/gpu/drm/vc4/vc4_plane.c

```diff
@@ -872,9 +881,7 @@ static int vc4_plane_mode_set(struct drm_plane *plane,
 			vc4_state->offsets[i] += src_y /
 						 (i ? v_subsample : 1) *
 						 tile_w;
-			vc4_state->offsets[i] += x_off /
-						 (i ? h_subsample : 1) *
-						 fb->format->cpp[i];
+			vc4_state->offsets[i] += x_off & ~(i ? 1:0);
```

Comment:
The lack of spaces around the colon is non-standard, but kernel Devs would probably prefer `~!!i`.